Report/Chapters/Design.tex

```diff
@@ -1,17 +1,103 @@
+There are multiple versions, ranging from a simple block that displays the string representation to a graph block which can show a function.
+
+Each kind of block has a different color to make them easily recognizable.
+
+\subsubsection{Errors}
+
+An important part of Viskell is representing programming errors.
+Instead of requiring users to compile a complete program, type errors are displayed as soon as a user connects two blocks.
+An error is represented as a colored line and colored type in the target block.
+An icon is displayed as well to help color blind users.
+
+\subsubsection{Menus}
+
+During the design process many ideas for the menus came up.
+One of the first ideas was to have a circular menu. \label{circular_menu}
+The advantages of the circle menu would be that the orientation of the device would not have any influence on the usability of the menu.
+This makes using the program on a large multi-touch table easier.
+However, the implementation of the menu would take too much time and the idea was therefore dismissed.
+
+The final design has two separate menus, one for adding new blocks and one for options on existing blocks.
+
+The primary menu uses `drawers' for the categories of the functions.
+This way it is easy to find functions even without an extensive knowledge of Haskell.
+In- and ontput blocks and the function definition block can be created using buttons at the bottom of the menu.
+This location is chosen to make them quickly available.
+The primary menu can be opened multiple times at once and moved around the screen so multiple users can work on a single program.
+
+The context menu is compact so is does not take up too much space.
+It contains clear icons which represent different actions.
+The icons are large enough to touch, even for people with large fingers.
+
+\subsubsection{Color scheme}
+
+The background color is a dark gray.
+This is chosen because it is less tiring for the eyes than white and allows for a high contrast.
+The background also incorporates a grid pattern so users can align elements on the screen if they like to.
+
+All other components feature a cheerful color scheme.
+This color scheme provides good contrast with the background and good recognizability for the different kinds of blocks.
+
+\subsection{Behaviour and interaction design}
+
 \section{Languages and libraries}
 
-% Java. Why?
-% JavaFX. Why?
-% TactileFX. Why?
+Viskell is written in Java using Java 8 with JavaFX.
+Java 8 is chosen because it, unlike Java 7, still receives public updates at the moment of writing.
+Language level 8 is chosen to make use of some of the new features in this version.
+
+Viskell uses TactileFX, a JavaFX touch screen library developed at the University of Twente.
+The client wanted the program to use this library, which also means that Viskell had to use Java and JavaFX.
+
+Furthermore, the following libraries have been used:
+
+\begin{itemize}
+	\item Antlr 4 (http://www.antlr.org/)
```

Comment:
There's a neat `\url` construction that turns these into hyperlinks in the resulting PDF.